File: C++/Sorting/selectionsort.cpp
#include <iostream>
using namespace std;


void selsort(int arr[], int size)
{
    int count = 0;

    for (int i = 0; i < size; i++)
    {
        int minindex = i;

        for (int j = i + 1; j < size; j++)
        {

            if (arr[minindex] > arr[j])
            {
                minindex = j;
            }
        }
        swap(arr[minindex], arr[i]);
        count++;

        // cout<<i<<" ";
        cout << arr[i] << " ";
    }

    cout << "\nCount: " <<count << endl;
}

int main()
{

    int arr[5] = {8, 2, 6, 1, 5};

    cout << "Sorted array : ";
    selsort(arr, 5);

    return 0;
}

File: Python/dsa/selectionSort.py


def selSort(arr):

    l = len(arr)
    count = 0

    for i in range(l):
        for j in range(i+1, l):

            print("i: ",arr[i], " ","j: ", arr[j])
            if arr[i] > arr[j]:
                temp = arr[i]
                arr[i] = arr[j]
                arr[j] = temp

                count+=1

        print(arr[i])

    return arr, count






arr = [ 40, 20, 10, 18, 22, 36]

print(selSort(arr))